### PR TITLE
feat(Color component): Appearance changed on colors component

### PR DESCRIPTION
**Pull request by josue2591**
## Summary

This PR tries to improve a little bit the appearance of the color component.

We made a similar build in a recent project, after showing it to some emulsify team members there was an interest for including this into compound.

## How to review this pull request
- [ ] After recompiling, check the [Color component ](http://localhost:6006/?path=/story/base-colors--palettes&globals=backgrounds.value:transparent)
- [ ] It should look as the screenshot above
Screenshot 2023-03-10 at 10 48 08 PM

### CODE REVIEW

**Review by codeb-a — COMMENTED** [review]
Hey @josue2591 leaving a review here. This looks great! I can confirm the colors palette is displaying. The only difference I see from the screen shot provided is that my background is showing a dark grey instead of a white background. Maybe this is a change from a different PR? (see screen shot).
Screenshot 2023-03-25 at 11 13 37 AM

**Comment by codeb-a** [comment]
> Hey @josue2591 leaving a review here. This looks great! I can confirm the colors palette is displaying. The only difference I see from the screen shot provided is that my background is showing a dark grey instead of a white background. Maybe this is a change from a different PR? (see screen shot).  

Just realized you probably have it in "light mode". You can ignore the previous comment.

**Review by codeb-a — APPROVED** [review]

File: components/00-base/01-colors/colors.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    const swatches = document.querySelectorAll('.cl-colors__item');
+    if (swatches) {
+      swatches.forEach((swatch) => {
+        const color = window
+          .getComputedStyle(swatch.querySelector('.cl-colors__swatch'), null)
+          .getPropertyValue('background-color');
+
+        const definition = swatch.querySelector('.cl-colors__definition');
+        console.log(swatch);
```
Comment (mcortes19) [inline components/00-base/01-colors/colors.js:28]: @josue2591 please remove this `console.log`

File: components/00-base/01-colors/colors.twig [hunk 2]
```diff
@@ -5,23 +5,28 @@
...
-    </ul>
-  {% endfor %}
+  <ul>
+    {% for palette in palettes %}
+      {% set palette_machine = palette.palette_heading|lower %}
+      {% include "@atoms/text/headings/_heading.twig" with {
+        heading_level: 3,
+        heading: palette.palette_heading,
+      } %}
+      <ul {{ bem('list', [palette_machine], colors_base_class) }} data-theme="{{ palette_machine }}">
```
Comment (mcortes19) [inline components/00-base/01-colors/colors.twig:15]: @josue2591 I think you need to add an `li` and inside it add the `ul` in order to pass [accessibility](https://www.w3.org/WAI/tutorials/page-structure/content/#nested-lists). You might also consider change the root `ul` (line 8) by `ol`

**Review by mcortes19 — CHANGES_REQUESTED** [review]
@josue2591 I left you couple of comments, let me know if you have doubts

**Comment by joetower** [comment]
@josue2591 This is a great improvement!

In addition to what Marvin mentioned, I'm wondering if we should add something like the following, if the user is using dark-mode.
```
.cl-colors__swatch {
  display: block;
  width: 11rem;
  height: 6rem;
  
  @media (prefers-color-scheme: dark) {
    border: 2px solid;
    box-shadow: 0px 4px 10px rgba(0,0,0, 0.27);
  }
}

```

My system, like Brandon's is using dark mode which, the way in which the swatches are rendering now, make the light mode `text` color #666666 swatch hard to differentiate from the background color in dark mode, which is also #666666. Here is a video demoing this addition.

https://user-images.githubusercontent.com/366413/236527747-4bc12645-511e-4f6b-8e86-9fc4c33721b0.mp4

File: components/00-base/01-colors/colors.twig (hunk 2 above)
Comment (josue2591) [reply components/00-base/01-colors/colors.twig]: Thanks! Changes applied

**Comment by josue2591** [comment]
Thanks @joetower I applied your suggestion but actually for any theme mode, because the issue was also for the light colors when the system mode was light.

@callinmullaney I think this is good to go

**Pull request merged**